core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2057,7 +2058,7 @@ private[spark] class DAGScheduler(
           // TODO: SPARK-35536: Cancel finalizeShuffleMerge if the stage is cancelled
           // TODO: during shuffleMergeFinalizeWaitSec
           shuffleClient.finalizeShuffleMerge(shuffleServiceLoc.host,
-            shuffleServiceLoc.port, shuffleId,
+            shuffleServiceLoc.port, shuffleId, stage.id,
```

Comment:
Replace `stage.id` with the shuffle sequence id

Reply:
Yeah haven't handle it yet. will handle it and update the code.